src/rust/engine/src/intrinsics/workspace_process.rs

```diff
@@ -0,0 +1,202 @@
+
+use std::env::current_dir;
+use std::path::Path;
+use std::process::Stdio;
+use std::str::FromStr;
+use std::sync::Arc;
+use std::time::Instant;
+
+use bytes::Bytes;
+use futures::future::{BoxFuture, FutureExt};
+use process_execution::local::{
+    apply_chroot, create_sandbox, prepare_workdir, setup_run_sh_script, CommandRunner,
+    KeepSandboxes,
+};
+use process_execution::ProcessExecutionStrategy;
+use pyo3::{IntoPy, PyAny, Python, ToPyObject};
+
+use crate::context::Context;
+use crate::externs;
+use crate::nodes::{ExecuteProcess, NodeResult, Snapshot};
+use crate::python::Value;
+
+pub(crate) fn workspace_process(
+    context: Context,
+    args: Vec<Value>,
+) -> BoxFuture<'static, NodeResult<Value>> {
+    log::debug!("workspace_process generating work unit closure");
+    // TODO: in_workunit!("workspace_process", Level::Debug, |_workunit| async move {
+    async move {
+        log::debug!("entering workspace_process");
+
+        let (py_workspace_process, py_process, process_config): (
+            Value,
+            Value,
+            externs::process::PyProcessExecutionEnvironment,
+        ) = Python::with_gil(|py| {
+            let py_workspace_process = (*args[0]).as_ref(py);
+            let py_process: Value = externs::getattr(py_workspace_process, "process").unwrap();
+            let process_config = (*args[1]).as_ref(py).extract().unwrap();
+            (
+                py_workspace_process.extract().unwrap(),
+                py_process,
+                process_config,
+            )
+        });
+        log::debug!("extracted py_workspace_process and friends");
+        match process_config.environment.strategy {
+            ProcessExecutionStrategy::Docker(_) | ProcessExecutionStrategy::RemoteExecution(_) => {
+                Err(format!(
+                    "Only local environments support running processes \
+               in the workspace, but a {} environment was used.",
+                    process_config.environment.strategy.strategy_type(),
+                ))
+            }
+            _ => Ok(()),
+        }?;
+        log::debug!("checked environment strategy");
+
+        let mut process =
+            ExecuteProcess::lift(&context.core.store(), py_process, process_config.clone())
+                .await?
+                .process;
+        log::debug!("process {process:?}");
+        let keep_sandboxes = Python::with_gil(|py| {
+            let py_interactive_process_obj = py_workspace_process.to_object(py);
+            let py_workspace_process = py_interactive_process_obj.as_ref(py);
+            let keep_sandboxes_value: &PyAny =
+                externs::getattr(py_workspace_process, "keep_sandboxes").unwrap();
+            KeepSandboxes::from_str(externs::getattr(keep_sandboxes_value, "value").unwrap())
+                .unwrap()
+        });
+        log::debug!("keep_sandboxes={keep_sandboxes:?}");
+
+        let mut tempdir = create_sandbox(
```

Comment:
It would be good to explain in a comment why we still have a sandbox, despite executing in the workspace, and how the process can interact with the sandbox, since it's not its CWD.